play-scala-secure-session-example/app/controllers/package.scala

```diff
@@ -29,12 +29,13 @@ package object controllers {
   object UserInfo {
     // Use a JSON format to automatically convert between case class and JsObject
     implicit val format: Format[UserInfo] = Json.format[UserInfo]
+    def unapply(userInfo: UserInfo): Option[(String)] = Some(userInfo.username)
   }
 
   val form = Form(
     mapping(
       "username" -> text
-    )(UserInfo.apply)(t => Some(t.toString()))
+    )(UserInfo.apply)(UserInfo.unapply)
```

Comment:
Scala 3 generates different unapply methods than Scala 2, so the nicer solution is to add back the same unapply method to the `UserInfo` companion object like Scala 2 generates.